include/aws/s3/s3_client.h

```diff
@@ -200,6 +200,20 @@ struct aws_s3_client_config {
     /* Callback and associated user data for when the client has completed its shutdown process. */
     aws_s3_client_shutdown_complete_callback_fn *shutdown_callback;
     void *shutdown_callback_user_data;
+
+    /*
+     * Optional.
+     * Configuration for using proxy from environment variable.
+     * Defaults to true
+     * Only works when proxy_options is not set.
+     */
+    struct aws_http_proxy_options *proxy_options;
```

Comment:
The meta request can be connected to a separate endpoint, maybe it makes sense to add the override of the proxy options from the meta request?